nevergrad/optimization/optimizerlib.py

```diff
@@ -227,10 +245,14 @@ def _internal_tell(self, x: ArrayLike, value: float) -> None:
     def _internal_provide_recommendation(self) -> ArrayLike:
         if self._es is None:
             raise RuntimeError("Either ask or tell method should have been called before")
-        if self.es.result.xbest is None:
-            return self.current_bests["pessimistic"].x
-        return self.es.result.xbest  # type: ignore
-
+        if self._fcmaes:
+            if self.es.best_x is None:
+                return self.current_bests["pessimistic"].x
+            return self.es.best_x  
+        else:
+            if self.es.result.xbest is None:
+                return self.current_bests["pessimistic"].x
+            return self.es.result.xbest  # type: ignore
```

Comment:
Less duplication is always better on the long run :D 
```suggestion
        cma_best = self.es.best_x if self._fcmaes else self.es.result.xbest
        if cma_best is None:
            return self.current_bests["pessimistic"].x
        return cma_best
```